pipelines/batch/src/main/java/com/google/fhir/analytics/ParquetMerger.java

```diff
@@ -246,6 +305,53 @@ public void processElement(ProcessContext c) {
               //   happens for TextIO. We should investigate this further and possibly file a bug.
               .withNumShards(options.getNumShards()));
     }
+    if (!dwhViews1.isEmpty() || !dwhViews2.isEmpty()) {
+      for (String viewName : dwhViews1) {
+        if (!dwhViews2.contains(viewName)) {
+          continue;
+        }
+        Pipeline pipeline = Pipeline.create(options);
+        pipelines.add(pipeline);
+        log.info("Merging materialized view {}", viewName);
+        ViewDefinition viewDef = viewMap.get(viewName);
+        PCollection<KV<String, Iterable<GenericRecord>>> groupedRecords =
+            readAndGroupById(
```

Comment:
In case of ViewDefinition parquet files, there are chances of having multiple records for the Id value, in the case grouping and just picking the latest value might mistakenly delete the intended records as well. Also, the logic of picking the latest value based on updateTimestamp might not be valid here as this column might not exist in the ViewDefinition.